app/assets/javascripts/components/user_profiles/course_details.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+
+
+const CourseDetails = createReactClass({
+  propTypes: {
+    courses: PropTypes.array
+  },
+
+  render() {
+    const elements = this.props.courses.map(course => {
+      return (
+        <a className="course" key={course.id} href={`/courses/${course.course_slug}`}>
```

Comment:
The `course` object here is not shaped the same way as the ones from course pages; it's actually derived from a CoursesUsers record, like this:

```
course_id: 201
course_school: "test"
course_slug: "test/Sci_Comm_brochure_yos_(now)"
course_term: "now"
course_title: "Sci Comm brochure yos"
user_count: 1
user_role: "Student"
```

To ensure unique keys, it should be something like:
```
`${course.course_slug}-${course.user_role}`
```